src/context.ts

```diff
@@ -257,12 +259,14 @@ type ResponseHeadersInit =
   | Record<string, string>
   | Headers
 
-interface ResponseInit {
+interface ResponseInit<T extends StatusCode = StatusCode> {
   headers?: ResponseHeadersInit
-  status?: number
+  status?: T
   statusText?: string
 }
 
+type ResponseOrInit<T extends StatusCode = StatusCode> = ResponseInit<T> | Response
```

Comment:
It has `Response | ResponseInit` because `ResponseInit` no longer satisfies the type of `Response`.

![CleanShot 2024-12-26 at 15 37 15@2x](https://github.com/user-attachments/assets/3fbca5d2-9824-4cfd-bbc4-5fe4f9211f5f)